File: practice/airport_traffic_controller/go/atc.go
package main

import (
	"sync"
	"fmt"
)

var lock = &sync.Mutex{}

type ATC struct {
}

var instance *ATC;

func GetInstance ()*ATC{
	if instance == nil {
		lock.Lock();
		defer lock.Unlock();
		instance = &ATC{}
	}
	return instance;
}


func main() {
	instance1:=GetInstance();
	instance2:=GetInstance();
	fmt.Println("This will print true if both the instaces are equal",instance1==instance2);
}

File: practice/airport_traffic_controller/py/atc.py
class ATC:
    '''
    Singleton Pattern Basic example using an ATC
    '''
    __instance = None
    
    def __new__(cls):
        if cls.__instance is None:
            cls.__instance= super().__new__(cls)
            cls.__instance = ATC()
        return cls.__instance
    
    def atc_print():
        print("This is the ATC")
    
#Usage
instance1=ATC()
instance2=ATC()

print("If instance1 is instance2 then will print True ",instance1 is instance2)